### ISSUE TITLE
scanner: mapping values are not allowed in this context

### ISSUE THREAD

**Issue by marcosfellipec** [realm/SwiftLint#3864]
### Step 1: Provide a summary of your problem
  * When use remote parent_config cause errors of parsing of yml file

### Step 2: Describe the problem:
error: YML Parsing Error: 141:34: error: scanner: mapping values are not allowed in this context:
    <span style="background-color: #79b8ff;width: 0%;" class="Progress-item progress-pjax-loader-bar "></span>
                                 ^
Could not read configuration: file Configuration.swift, line 240
/Users/costamn/Library/Developer/Xcode/DerivedData/BTGBankingFeatures-aixgnaghtchjnldzvnmdofiidvbl/Build/Intermediates.noindex/TestApp.build/Debug-iphonesimulator/TestApp.build/Script-FAEC793025B25337001D5587.sh: line 2: 46911 Abort trap: 6           "$PODS_ROOT/SwiftLint/swiftlint" --config "$SRCROOT/.swiftlint.yml"

#### REMOTE YML:
  ```

warning_threshold: 30

nesting: 
  type_level: 3
  function_level: 6

force_cast:
  severity: error

force_unwrapping:
  severity: error

cyclomatic_complexity:
  ignores_case_statements: true
  warning: 8
  error: 10

function_parameter_count:
  warning: 6
  error: 8

file_length:
  error: 500
  ignore_comment_only_lines: true

type_body_length:
  warning: 300
  error: 400

function_body_length:
  warning: 40
  error: 80

line_length:
  warning: 180
  error: 200
  ignores_comments: true
  ignores_function_declarations: true
  ignores_interpolated_strings: true

type_name:
  severity: warning
  validates_start_with_lowercase: false
  max_length:
    warning: 50

identifier_name:
  allowed_symbols:
    - _
  min_length:
    warning: 2
    error: 1
  max_length:
    warning: 40
    error: 60
  validates_start_with_lowercase: false
  excluded:
    - x
    - y
    - i

disabled_rules:
  - inclusive_language
  - let_var_whitespace

opt_in_rules:
  - closing_brace
  - closure_end_indentation
  - collection_alignment
  - colon
  - comma
  - contains_over_first_not_nil
  - empty_count
  - empty_string
  - explicit_init
  - explicit_self
  - for_where
  - force_unwrapping
  - force_try
  - let_var_whitespace
  - mark
  - multiline_parameters
  - overridden_super_call
  - redundant_nil_coalescing
  - return_arrow_whitespace
  - switch_case_on_newline
  - toggle_bool
  - unneeded_parentheses_in_closure_argument
  - unused_import
  - vertical_parameter_alignment_on_call
  - yoda_condition

custom_rules:
  comments_space:
    name: "Space After Comment"
    regex: "(^ *//W+)"
    message: "There should be a space after //"
    severity: warning

  empty_line_after_mark:
    name: "Empty Line After MARK"
    regex: "(^ *\/\/ MARK: -[ a-zA-Z0-9=?.\\(\\)\\{\\}:,><!]*\n *(?!(?:\\}))\\S+)"
    message: "There should be an empty line after a marks"
    severity: warning

  empty_line_before_mark:
    name: "Empty Line Before MARK"
    regex: "(?!(?:\\}))\\S+\n *\/\/ MARK: -[ a-zA-Z0-9=?.\\(\\)\\{\\}:,><!]*"
    message: "There should be an empty line before a marks"
    severity: warning

  no_prints:
    name: 'Forbidden print'
    regex: 'print[(]'
    message: "Don't forget prints into code"
    severity: warning
  
  numbers_smell:
    name: 'Numbers smell'
    regex: '(return |case |\w\(|: |\?\? |\, |== |<=? |>=? |\+= |\-= |\/= |\*= |%= |\w\.\w+ = )\(*-?\d{2,}'
    message: 'Instead, numbers need to be defined as constants'
    severity: warning
  
  multiline_commented_code:
    name: 'Multiline Commented Code'
    regex: '^\s*[a-z]'
    match_kinds: comment
    message: "Comment starting with lowercase letter - did you forget to delete old code?"
    severity: warning

  commented_code:
    name: 'Commented Code'
    regex: '(?<!:|\/)\/\/\h*[a-z.](?!wiftlint)'
    message: "Comment starting with lowercase letter - did you forget to delete old code?"
    severity: warning
  ```
#### LOCAL YML:
  ```
parent_config: https://github.com/apptest/apptest/Classes/Danger/.swiftlint.yml
  ```

**Comment by marcelofabri** [comment]
It looks like your remove configuration file is not valid YML

**Comment by marcosfellipec** [comment]
> It looks like your remove configuration file is not valid YML

@marcelofabri 
Work when used like a local file

**Comment by marcelofabri** [comment]
Can you paste the actual file here? The snippet that you added is definitely mal formed YML

**Comment by uhooi** [comment]
I get the same error as well.

__ERROR MSG:__

```
Showing Recent Messages
Could not read configuration: file Configuration.swift, line 240

/Users/uhooi/Library/Developer/Xcode/DerivedData/UhooiPicBook-xxxxxxxxxxxxxxxxxxxxxxxxxx/Build/Intermediates.noindex/Develop.build/Debug-iphonesimulator/UhooiPicBook.build/Script-999999999999999999999999.sh: line 23: 44780 Abort trap: 6           mint run realm/SwiftLint swiftlint --config $SRCROOT/../.swiftlint.yml

scanner: mapping values are not allowed in this context:
```

__REMOTE YML:__

```yaml
disabled_rules:
  #- block_based_kvo
  #- class_delegate_protocol
  #- closing_brace
  #- closure_parameter_position
  #- colon
  #- comma
  #- comma_inheritance
  #- comment_spacing
  #- compiler_protocol_init
  #- computed_accessors_order
  #- control_statement
  #- custom_rules
  #- cyclomatic_complexity
  #- deployment_target
  #- discouraged_direct_init
  #- duplicate_enum_cases
  #- duplicate_imports
  #- duplicated_key_in_dictionary_literal
  #- dynamic_inline
  #- empty_enum_arguments
  #- empty_parameters
  #- empty_parentheses_with_trailing_closure
  #- file_length
  #- for_where
  #- force_cast
  #- force_try
  #- function_body_length
  #- function_parameter_count
  #- generic_type_name
  #- identifier_name
  #- implicit_getter
  #- inclusive_language
  #- inert_defer
  #- is_disjoint
  #- large_tuple
  #- leading_whitespace
  #- legacy_cggeometry_functions
  #- legacy_constant
  #- legacy_constructor
  #- legacy_hashing
  #- legacy_nsgeometry_functions
  #- legacy_random
  #- line_length
  #- mark
  #- multiple_closures_with_trailing_closure
  #- nesting
  #- no_fallthrough_only
  #- no_space_in_method_call
  #- notification_center_detachment
  #- ns_number_init_as_function_reference
  #- nsobject_prefer_isequal
  #- opening_brace
  #- operator_whitespace
  #- orphaned_doc_comment
  #- private_over_fileprivate
  #- private_unit_test
  #- protocol_property_accessors_order
  #- reduce_boolean
  #- redundant_discardable_let
  #- redundant_objc_attribute
  #- redundant_optional_initialization
  #- redundant_set_access_control
  #- redundant_string_enum_value
  #- redundant_void_return
  #- return_arrow_whitespace
  #- self_in_property_initialization
  #- shorthand_operator
  #- statement_position
  #- superfluous_disable_command
  #- switch_case_alignment
  #- syntactic_sugar
  #- todo
  #- trailing_comma
  #- trailing_newline
  #- trailing_semicolon
  #- trailing_whitespace
  #- type_body_length
  #- type_name
  #- unavailable_condition
  #- unneeded_break_in_switch
  #- unused_capture_list
  #- unused_closure_parameter
  #- unused_control_flow_label
  #- unused_enumerated
  #- unused_optional_binding
  #- unused_setter_value
  #- valid_ibinspectable
  #- vertical_parameter_alignment
  #- vertical_whitespace
  #- void_return
  #- void_function_in_ternary
  #- xctfail_message

opt_in_rules:
  - accessibility_label_for_image
  - accessibility_trait_for_button
  #- anonymous_argument_in_multiline_closure
  - array_init
  - attributes
  - balanced_xctest_lifecycle
  - closure_body_length
  - closure_end_indentation
  - closure_spacing
  - collection_alignment
  - comma_inheritance
  - conditional_returns_on_newline
  - contains_over_filter_count
  - contains_over_filter_is_empty
  - contains_over_first_not_nil
  - contains_over_range_nil_comparison
  - convenience_type
  - discarded_notification_center_observer
  - discouraged_assert
  - discouraged_object_literal
  - discouraged_optional_boolean
  - discouraged_optional_collection
  - empty_collection_literal
  - empty_count
  - empty_string
  - empty_xctest_method
  - enum_case_associated_values_count
  - expiring_todo
  #- explicit_acl # できる限りACLを省略したいため
  #- explicit_enum_raw_value # ローバリューを省略することもあるため
  - explicit_init
  #- explicit_top_level_acl # できる限りACLを省略したいため
  #- explicit_type_interface # できる限り型推論したいため
  #- extension_access_modifier # このルールの意味を理解していないため
  - fallthrough
  - fatal_error_message
  #- file_header # このルールの意味を理解していないため
  - file_name
  - file_name_no_space
  - file_types_order
  - first_where
  - flatmap_over_map_reduce
  - force_unwrapping
  - function_default_parameter_at_end
  - ibinspectable_in_extension
  - identical_operands
  - implicit_return
  #- implicitly_unwrapped_optional # VIPERで変数を `!` で定義したいため
  #- indentation_width # `guard` のインデントに対応していなく、メソッドの引数でも引っかかって厳しいため
  - joined_default_parameter
  - last_where
  - legacy_multiple
  #- let_var_whitespace # 空白行を設けたくないこともあるため
  - literal_expression_end_indentation
  - lower_acl_than_parent
  - missing_docs
  - modifier_order
  #- multiline_arguments # 引数は同じ行に2つ入れたいこともあるため
  #- multiline_arguments_brackets # 括弧で行を増やしたくないため
  #- multiline_function_chains # 関数の呼び出しは同じ行に2つ入れたいこともあるため
  #- multiline_literal_brackets # 括弧で行を増やしたくないため
  #- multiline_parameters # 引数は同じ行に2つ入れたいこともあるため
  #- multiline_parameters_brackets # 括弧で行を増やしたくないため
  - nimble_operator
  #- no_extension_access_modifier # エクステンションにACLを設定したいことがあるため
  #- no_grouping_extension # エクステンションでグルーピングしたいことがあるため
  #- non_private_xctest_member # TODO: 未リリースのため
  - nslocalizedstring_key
  - nslocalizedstring_require_bundle
  #- number_separator # 数字を `_` で区切りたくないため
  #- object_literal # リテラルで生成したくないこともあるため
  - operator_usage_whitespace
  - optional_enum_case_matching
  - overridden_super_call
  - override_in_extension
  - pattern_matching_keywords
  #- prefer_nimble # Nimbleを使っていないため
  - prefer_self_in_static_references
  - prefer_self_type_over_type_of_self
  - prefer_zero_over_explicit_init
  #- prefixed_toplevel_constant # 定数のプリフィックスに `k` を付けたくないため
  - private_action
  - private_outlet
  - private_subject
  #- prohibited_interface_builder # IBを使ってビューを生成したいため
  - prohibited_super_call
  - quick_discouraged_call
  - quick_discouraged_focused_test
  - quick_discouraged_pending_test
  - raw_value_for_camel_cased_codable_enum
  - reduce_into
  - redundant_nil_coalescing
  - redundant_type_annotation
  #- required_deinit # できる限りデイニシャライザを省略したいため
  - required_enum_case
  - return_value_from_void_function
  - self_binding
  - shorthand_optional_binding
  - single_test_class
  - sorted_first_last
  #- sorted_imports # インポート文をアルファベット順以外に並び替えたいこともあるため
  - static_operator
  - strict_fileprivate
  #- strong_iboutlet # `@IBOutlet` を `weak` で定義したいため
  - switch_case_on_newline
  - test_case_accessibility
  - toggle_bool
  - trailing_closure
  - type_contents_order
  - unavailable_function
  - unneeded_parentheses_in_closure_argument
  - unowned_variable_capture
  - untyped_error_in_catch
  - vertical_parameter_alignment_on_call
  #- vertical_whitespace_between_cases # Switch文のケース間に空白行を設けたくないため
  #- vertical_whitespace_closing_braces # 中括弧を閉じる前に空白行を設けたいことがあるため
  #- vertical_whitespace_opening_braces # 中括弧を開く前に空白行を設けたいことがあるため
  - xct_specific_matcher
  - yoda_condition

analyzer_rules:
  - capture_variable
  #- explicit_self # 関数は `self.` を付けずに呼び出したいため
  - typesafe_array_init
  - unused_declaration
  - unused_import

line_length:
  warning: 300
  error: 500

identifier_name:
  min_length: 1 # `r` `g` `b` などを使いたいため
```

__LOCAL YML:__

```yaml
parent_config: https://github.com/uhooi/SwiftLint-Config/blob/feature/0.50.0/uhooi-base-swiftlint-config.yml

disabled_rules:
  - missing_docs

included:
  - Shared
  - UhooiPicBook
  - UhooiPicBookStickers
  - UhooiPicBookWidgets
  - UhooiPicBookWidgetsConfigurableIntent
  - Sources

excluded:
  - UhooiPicBookTests
  - UhooiPicBookUITests
  - Tests
  - SourcePackages
  - Tools
  - Generated
```

__SwiftLint version:__ `0.50.0`

**Comment by uhooi** [comment]
Sorry, I made a mistake on the YAML path.

__LOCAL YML:__

```diff
- parent_config: https://github.com/uhooi/SwiftLint-Config/blob/feature/0.50.0/uhooi-base-swiftlint-config.yml
+ parent_config: https://raw.githubusercontent.com/uhooi/SwiftLint-Config/feature/0.50.0/uhooi-base-swiftlint-config.yml
```

**Comment by SimplyDanny** [comment]
I checked with [this config file](https://raw.githubusercontent.com/uhooi/SwiftLint-Config/main/uhooi-base-swiftlint-config.yml) in 0.50.3 and it works well on my side. Do you confirm this is the right configuration and the problem still exists for you, @uhooi?

**Comment by uhooi** [comment]
@SimplyDanny 
I too confirm that there is no problem. Thank you for your time.

**Comment by SimplyDanny** [comment]
As it seems the problem also doesn't exist for the original creator of this issue anymore. Thus I'm going to close it.
